Add a python code to test whether the model is correct or not

J3y0 opened the pull request:
In this PR, I would bring a **python code** with which we can play around with some tests **in order to test** whether the mathematical model is *correct* or *not*.

I also fixed some typos in the `README.md` file.

J3y0 committed:
Correct a few typos

J3y0 committed:
Create a python code to solve the defined problem

First of all, I have written a function which creates the weight matrix
according to the ranking matrix given as an input.

A ranking matrix is defined as follow:
- Lines represent each subject's ranking
- Columns are the ranking itself (ex: the 1st column represents the 1st
  choice of the subject)

Secondly, I create a temporary function named `solve_problem`. By
temporary, I mean the format could be better, either creating another
file either significantly improving the structure of the function.

**IMPORTANT: **This python code has only test purposes ! It only
**asserts** the mathematical model is correct !

Turtyo commented on sorter.py:
```diff
@@ -0,0 +1,99 @@
+    return: a weight matrix
+    """
+    w = []
+    n, m = np.array(ranking).shape
```
why define m again if it is given as an argument ?

Turtyo commented on sorter.py:
```diff
@@ -0,0 +1,99 @@
+    """
+    w = []
+    n, m = np.array(ranking).shape
+    _Vmax = 10 # Arbitrary value for now
```
it would probably be better to have those values in another file, this way it's easier to edit

Turtyo commented on sorter.py:
```diff
@@ -0,0 +1,99 @@
+    ## Constraints
+    # Each subject must have k activities
+    for i in range(n):
+        prob += pulp.lpSum(a[i,:]) == k
```
maybe put a comment to explain the definition of a, or define a before this

Turtyo commented on sorter.py:
```diff
@@ -0,0 +1,99 @@
+            [1, 3, 2],
+            [3, 1, 2]]
+# n is the number of subject, m the number of activities
+n,m = np.array(ranking).shape
```
same as line 14

Turtyo commented on sorter.py:
```diff
@@ -0,0 +1,99 @@
+a = np.empty((n,m), dtype=object)
+for i in range(n):
+    for j in range(m):
+        a[i,j] = pulp.LpVariable('a_{}_{}'.format(i,j),cat=pulp.LpBinary)
```
maybe using an f-string instead of a .format() would be better

Turtyo reviewed (changes requested):
Finished the review
We will need to test limit cases (in term of size or impossible configurations)

J3y0 committed:
Change `sorter.py` according to the review

A few more details about what changed:
- The values for the exponential law are now stored
in a json file.
- Redefining m (the number of activities) correctly
- Slight change on the f-string

Turtyo reviewed (approved):
all clear, thank you for the changes !

J3y0 committed:
Correct an *out of bound* error
